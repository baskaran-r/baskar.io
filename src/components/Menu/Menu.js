// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import classNames from 'classnames/bind';
import styles from './Menu.module.scss';
import { useSiteMetadata } from '../../hooks';

type Props = {|
  +horizontal?: boolean,
  +noMargin?: boolean,
  +bold?: boolean,
|};

const cx = classNames.bind(styles);

const Menu = ({ horizontal, noMargin }: Props) => {
  const { menu } = useSiteMetadata();

  return (
    <nav
      className={cx({
        menu: true,
        horizontal,
        'no-margin': noMargin,
      })}
    >
      <ul className={styles['menu__list']}>
        {menu.map((item) => (
          <li className={styles['menu__list-item']} key={item.path}>
            <Link
              to={item.path}
              className={styles['menu__list-item-link']}
              activeClassName={styles['menu__list-item-link--active']}
            >
              {item.label}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Menu;
