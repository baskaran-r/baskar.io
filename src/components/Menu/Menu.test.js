// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';

import Menu from './Menu';

describe('Menu', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => render(siteMetadata),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Menu />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
