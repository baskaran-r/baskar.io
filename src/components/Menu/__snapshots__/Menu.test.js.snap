// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<nav
  className="menu"
>
  <ul
    className="menu__list"
  >
    <li
      className="menu__list-item"
    >
      <a
        className="menu__list-item-link"
        href="/test/1/"
      >
        Test label 1
      </a>
    </li>
    <li
      className="menu__list-item"
    >
      <a
        className="menu__list-item-link"
        href="/test/2/"
      >
        Test label 2
      </a>
    </li>
    <li
      className="menu__list-item"
    >
      <a
        className="menu__list-item-link"
        href="/test/3/"
      >
        Test label 3
      </a>
    </li>
  </ul>
</nav>
`;
